tests/end-to-end/external-solution-integration/testkit/connector.go

```diff
@@ -0,0 +1,157 @@
+package testkit
+
+import (
+	"bytes"
+	"crypto/tls"
+	"crypto/x509"
+	"encoding/base64"
+	"encoding/json"
+	"net/http"
+
+	"github.com/kyma-project/kyma/tests/end-to-end/external-solution-integration/resourceskit"
+	"github.com/kyma-project/kyma/tests/end-to-end/external-solution-integration/wait"
+	"github.com/sirupsen/logrus"
+)
+
+type ConnectorClient interface {
+	GetToken() (string, error)
+	GetInfo(url string) (*InfoResponse, error)
+	GetCertificate(url string, csr []byte) ([]*x509.Certificate, error)
+}
+
+type connectorClient struct {
+	trClient   resourceskit.TokenRequestClient
+	httpClient *http.Client
+	logger     logrus.FieldLogger
+}
+
+func NewConnectorClient(trClient resourceskit.TokenRequestClient, skipVerify bool, logger logrus.FieldLogger) ConnectorClient {
+	return &connectorClient{
+		trClient:   trClient,
+		httpClient: newHttpClient(skipVerify),
+		logger:     logger,
+	}
+}
+
+func newHttpClient(skipVerify bool) *http.Client {
+	tr := &http.Transport{
+		TLSClientConfig: &tls.Config{InsecureSkipVerify: skipVerify},
+	}
+	client := &http.Client{Transport: tr}
+	return client
+}
+
+func (cc *connectorClient) GetToken() (string, error) {
+	_, err := cc.trClient.CreateTokenRequest()
+	if err != nil {
+		cc.logger.Error(err)
+		return "", err
+	}
+
+	err = wait.Until(5, 10, cc.isTokenRequestReady)
```

Comment:
I think that implementation using `watch` would be better here and in other places waiting for k8s resources. First of all, it would be faster.